src/silhouetteplot.jl

```diff
@@ -0,0 +1,83 @@
+"""
+	silhouetteplot(C,X[,D];...)
+Make a silhouette plot to assess the quality of a clustering. `C` must be a `ClusteringResult` (see the `Clustering` package), and `X` is a matrix in which each column represents a data point. If supplied, `D` should be a distance matrix (as in `Distances`); otherwise, pairwise Euclidean distances are used.
+
+Each data point has a silhouette score between -1 and 1 indicating how unambiguously the point belongs to its assigned cluster. These are sorted within each cluster and portrayed using horizontal bars. Also shown is a dashed line at the average score. Typically a high-quality clustering has significant numbers of bars within each cluster that cross the line, and few negative scores overall.
+
+See also: [`Clustering.silhouettes`](@ref), [`Distances`](@ref).
+
+# Examples
+
+```
+using Clustering, Distances, Plots
+# random dataset with 3-ish clusters in 5 dimensions
+X = hcat([rand(5,1) .+ 0.2*randn(5, 200) for _=1:3]...)
+D = pairwise(Euclidean(),X,dims=2)
+R = kmeans(D, 3; maxiter=200, display=:iter)
+
+silhouetteplot(R,X,D)
+```
+"""
+silhouetteplot
+
+@userplot SilhouettePlot
+@recipe function f(h::SilhouettePlot)#R::ClusteringResult,X::AbstractArray,D::AbstractMatrix=[];distance=Euclidean())
+	narg = length(h.args)
+	@assert narg > 1 "At least two arguments are required."
+	R = h.args[1]
+	@assert R isa ClusteringResult "First argument must be a ClusteringResult."
+	X = h.args[2]
+	@assert X isa AbstractArray "Second argument must be an array."
+	if narg > 2
+		D = h.args[3]
+		@assert D isa AbstractMatrix "Third argument must be a distance matrix."
+	else
+		D = pairwise(Euclidean(),X,dims=2)
+	end
+
+	a = assignments(R)  # assignments to clusters
+	c = counts(R)  # cluster sizes
+	k = length(c)  # number of clusters
+	n = sum(c)     # number of points overall
+
+	s = silhouettes(R,D)
+
+	# Settings for the axes
+	legend --> false
+	yflip := true
```

Comment:
Why do you override settings for these three attributes? Might `-->` not be better?

Reply:
`yflip` needs to be known in order to get the sorting order right. I'm not sure how to read that attribute if it wasn't set explicitly, and I thought top-to-bottom was most natural.

`xlims` and `yticks` should probably be "soft" assignments, I agree. Will modify.

Reply:
Actually, I've learned that if I don't assign `yticks:=`, the labels don't get preserved. I'd really rather show the size of each cluster than the cumulative sizes. Do you know a fix?

Reply:
You can get the value by `plotattributes[:yticks]`, I think.  You can also read it, and then explicitly assign what you read to it.

Reply:
No, if I try to reference `plotattributes[:yticks]` when it was left unset in the invoking plot command, I get a "key not found" error. I suppose that unset attributes aren't given values until later in the pipeline.

Reply:
I've just avoided the issue by making `yflip` a keyword setting in `silhouetteplot`. It's true by default, but setting it to false is respected.